tests/backends/base/test_schema.py

```diff
@@ -0,0 +1,22 @@
+from django.db import connection, models
+from django.test import TestCase
+
+
+class SchemaEditorTests(TestCase):
+
+    def test_effective_default_callable(self):
+        """SchemaEditor.effective_default() shouldn't call callable defaults."""
+        class MyStr(str):
+            def __call__(self):
+                return self
+
+        class MyCharField(models.CharField):
+            def _get_default(self):
+                return self.default
+
+            def get_db_prep_save(self, default, connection):
+                return default
+
+        field = MyCharField(max_length=1, default=MyStr)
+        with connection.schema_editor() as editor:
+            self.assertEqual(editor.effective_default(field), MyStr)
```

Comment:
That makes me wonder if making `effective_default` a class method and allow passing of `connection` would make it more easily testable. It's kind of a shame that we have to create a database connection and transactions to test this simple method.

Reply:
Does the second commit appease you? ;-)